src/lib/database.js

```diff
@@ -72,102 +73,213 @@ export function buildDatetimeQuery(parameters) {
+    const operatorsObject = query[property]
+    const operators = Object.keys(operatorsObject)
 
-  let filterQueries = []
-  let mustNotQueries = []
+    // eq
+    if (operators.includes(eq)) {
+      accumulator.push({
+        term: {
+          [`properties.${property}`]: operatorsObject.eq
+        }
+      })
+    }
 
-  if (query) {
-    // Using reduce rather than map as we don't currently support all
-    // stac query operators.
-    filterQueries = Object.keys(query).reduce((accumulator, property) => {
-      const operatorsObject = query[property]
-      const operators = Object.keys(operatorsObject)
-
-      // eq
-      if (operators.includes(eq)) {
-        accumulator.push({
-          term: {
-            [`properties.${property}`]: operatorsObject.eq
+    // in
+    if (operators.includes(inop)) {
+      accumulator.push({
+        terms: {
+          [`properties.${property}`]: operatorsObject.in
+        }
+      })
+    }
+
+    // startsWith
+    if (operators.includes(startsWith)) {
+      accumulator.push({
+        prefix: {
+          [`properties.${property}`]: {
+            value: operatorsObject.startsWith
           }
-        })
-      }
+        }
+      })
+    }
 
-      // in
-      if (operators.includes(inop)) {
-        accumulator.push({
-          terms: {
-            [`properties.${property}`]: operatorsObject.in
+    // endsWith
+    if (operators.includes(endsWith)) {
+      accumulator.push({
+        wildcard: {
+          [`properties.${property}`]: {
+            value: `*${operatorsObject.endsWith}`
           }
-        })
-      }
+        }
+      })
+    }
 
-      // startsWith
-      if (operators.includes(startsWith)) {
-        accumulator.push({
-          prefix: {
-            [`properties.${property}`]: {
-              value: operatorsObject.startsWith
-            }
+    // contains
+    if (operators.includes(contains)) {
+      accumulator.push({
+        wildcard: {
+          [`properties.${property}`]: {
+            value: `*${operatorsObject.contains}*`
           }
-        })
+        }
+      })
+    }
+
+    // lt, lte, gt, gte
+    const rangeQuery = buildRangeQuery(property, operators, operatorsObject)
+    if (rangeQuery) {
+      accumulator.push(rangeQuery)
+    }
+
+    return accumulator
+  }, filterQueries)
+
+  const neq = 'neq'
+  let mustNotQueries = []
+
+  mustNotQueries = Object.keys(query).reduce((accumulator, property) => {
+    const operatorsObject = query[property]
+    const operators = Object.keys(operatorsObject)
+
+    // neq
+    if (operators.includes(neq)) {
+      accumulator.push({
+        term: {
+          [`properties.${property}`]: operatorsObject.neq
+        }
+      })
+    }
+
+    return accumulator
+  }, mustNotQueries)
+
+  return {
+    bool: {
+      filter: filterQueries,
+      must_not: mustNotQueries
+    }
+  }
+}
+
+function buildFilterExtQuery(filter) {
+  const LogicalOp = {
```

Comment:
these should be defined outside the function.

Reply:
Done.